.config/nvim/ftplugin/java.lua

```diff
@@ -1,142 +1,159 @@
 -- JDTLS (Java LSP) configuration
-local jdtls = require('jdtls')
-local project_name = vim.fn.fnamemodify(vim.fn.getcwd(), ':p:h:t')
-local workspace_dir = vim.env.HOME .. '/jdtls-workspace/' .. project_name
+local home = vim.env.HOME -- Get the home directory
+
+local jdtls = require("jdtls")
+local project_name = vim.fn.fnamemodify(vim.fn.getcwd(), ":p:h:t")
+local workspace_dir = home .. "/jdtls-workspace/" .. project_name
+
+-- Determine OS
+-- Use this option to avoid explicitly adding manually everytime you switch OS e.g "config_linux", "config_mac"
+-- This saves a lot of time if you are working with multiple OS e.g Linux or Mac OS
+if vim.fn.has("mac") == 1 then
+	CONFIG = "mac"
+elseif vim.fn.has("unix") == 1 then
+	CONFIG = "linux"
+else
+	print("Unsupported system")
+end
 
 -- Needed for debugging
 local bundles = {
-  vim.fn.glob(vim.env.HOME .. '/.local/share/nvim/mason/share/java-debug-adapter/com.microsoft.java.debug.plugin.jar'),
+	vim.fn.glob(home .. "/.local/share/nvim/mason/share/java-debug-adapter/com.microsoft.java.debug.plugin.jar"),
 }
 
 -- Needed for running/debugging unit tests
-vim.list_extend(bundles, vim.split(vim.fn.glob(vim.env.HOME .. "/.local/share/nvim/mason/share/java-test/*.jar", 1), "\n"))
+vim.list_extend(bundles, vim.split(vim.fn.glob(home .. "/.local/share/nvim/mason/share/java-test/*.jar", 1), "\n"))
 
 -- See `:help vim.lsp.start_client` for an overview of the supported `config` options.
 local config = {
-  -- The command that starts the language server
```

Comment:
Sorry, another nit-picky comment, but can you please remove code formatting and quotation changes from the PR.  It makes the review very difficult and my formatting options may be different than your formatting options which means it will get reverted again on my next commit anyway.